pkg/rke2/registries.go

```diff
@@ -69,24 +69,38 @@ func GenerateRegistries(rke2ConfigRegistry RegistryScope) (*Registry, []bootstra
 				return &Registry{}, []bootstrapv1.File{}, err
 			}
 
-			for _, secretEntry := range []string{"tls.crt", "tls.key", "ca.crt"} {
-				if tlsSecret.Data[secretEntry] == nil {
-					rke2ConfigRegistry.Logger.Error(err, "TLS Secret for the registry is missing entries!", "secret-missing-entry", secretEntry)
+			isCert := tlsSecret.Data["ca.crt"] != nil
+			isSkipVerify := tlsSecret.Data["insecureSkipVerify"] != nil
 
-					return &Registry{}, []bootstrapv1.File{}, err
-				}
+			ok := isCert || isSkipVerify
+
+			if !ok {
+				rke2ConfigRegistry.Logger.Error(
+					err,
+					"TLS Secret for the registry is missing entries! Possible entries are: \"ca.crt\" (with optional: \"tls.crt\" and \"tls.key\") OR \"insecureSkipVerify\"",
+					"secret-entries", bsutil.GetMapKeysAsString(tlsSecret.Data))
 
-				files = append(files, bootstrapv1.File{
-					Path:    registryCertsPath + "/" + secretEntry,
-					Content: string(tlsSecret.Data[secretEntry]),
-				})
+				return &Registry{}, []bootstrapv1.File{}, err
 			}
 
-			registryConfig.TLS = &TLSConfig{
-				InsecureSkipVerify: regConfig.TLS.InsecureSkipVerify,
-				CAFile:             registryCertsPath + "/" + "ca.crt",
-				CertFile:           registryCertsPath + "/" + "tls.crt",
-				KeyFile:            registryCertsPath + "/" + "tls.key",
+			for _, secretEntry := range []string{"tls.crt", "tls.key", "ca.crt"} {
```

Comment:
I'd like to recommend making "tls.crt", "tls.key", "ca.crt" constants so we don't retype the same strings if needed.

Reply:
changed!